### PR TITLE
gh-110905: [Enum] minor fixes and cleanup

### PR DESCRIPTION
**Pull request by eendebakpt**
* Fix `enum._is_private` (bug + anti-bug) for triple underscore names
* Rewrite `_is_sunder` check in the same style as `_is_private`
* Remove unused variable `blanks`
* Update docstring

* Issue: gh-110905

### CODE REVIEW

File: Lib/enum.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def _is_private(cls_name, name):
     if (
             len(name) > pat_len
             and name.startswith(pattern)
-            and name[pat_len:pat_len+1] != ['_']
+            and name[pat_len] != '_'
```
Comment (ethanfurman) [inline Lib/enum.py]: Just remove this line.
```suggestion
```

File: Lib/test/test_enum.py [hunk 2]
```diff
@@ -173,7 +173,7 @@ class TestHelpers(unittest.TestCase):
     dunder_names = '__mal__', '__bien__', '__que_que__'
     private_names = '_MyEnum__private', '_MyEnum__still_private'
```
Comment (ethanfurman) [inline Lib/test/test_enum.py]:
```suggestion
    private_names = '_MyEnum__private', '_MyEnum__still_private', '_MyEnum___triple_private'
```

File: Lib/test/test_enum.py [hunk 3]
```diff
@@ -173,7 +173,7 @@ class TestHelpers(unittest.TestCase):
...
     private_and_sunder_names = '_MyEnum__private_', '_MyEnum__also_private_'
-    random_names = 'okay', '_semi_private', '_weird__', '_MyEnum__'
+    random_names = 'okay', '_semi_private', '_weird__', '_MyEnum__', '___triple_underscore'
```
Comment (ethanfurman) [inline Lib/test/test_enum.py]: Move this up to the `private_names` line.
```suggestion
    random_names = 'okay', '_semi_private', '_weird__', '_MyEnum__',
```

**Review by ethanfurman — CHANGES_REQUESTED** [review]
A couple changes and we're good to go!

File: Lib/test/test_enum.py [hunk 4]
```diff
@@ -171,9 +171,9 @@ class TestHelpers(unittest.TestCase):
 
     sunder_names = '_bad_', '_good_', '_what_ho_'
     dunder_names = '__mal__', '__bien__', '__que_que__'
-    private_names = '_MyEnum__private', '_MyEnum__still_private'
+    private_names = '_MyEnum__private', '_MyEnum__still_private', '_MyEnum___triple_private'
     private_and_sunder_names = '_MyEnum__private_', '_MyEnum__also_private_'
-    random_names = 'okay', '_semi_private', '_weird__', '_MyEnum__'
+    random_names = 'okay', '_semi_private', '_weird__', '_MyEnum__',
```
Comment (eendebakpt) [inline Lib/test/test_enum.py]:
```suggestion
    random_names = 'okay', '_semi_private', '_weird__', '_MyEnum__'
```

**Review by ethanfurman — APPROVED** [review]

**Comment by sunmy2019** [comment]
@ethanfurman shall we backport this?

**Pull request merged**